crates/experimentation_platform/src/api/experiments/cac_api.rs

```diff
@@ -0,0 +1,228 @@
+    let http_client = reqwest::Client::new();
+    let url = state.cac_host.clone() + "/config/resolve";
+    let mut exp_context_dimension_value = extract_dimensions(exp_context)?;
+    exp_context_dimension_value.insert(
+        "context_id".to_string(),
+        Value::String(context_id.to_string()),
+    );
+
+    let user_str = serde_json::to_string(user).map_err(|err| {
+        log::error!("Something went wrong, failed to stringify user data {err}");
+        unexpected_error!(
+            "Something went wrong, failed to stringify user data {}",
+            err
+        )
+    })?;
+
+    let extra_headers = vec![("x-user", user_str)];
+
+    let headers_map = construct_header_map(
+        &workspace_request.workspace_id,
+        &workspace_request.organisation_id,
+        extra_headers,
+    )?;
+    let response = http_client
+        .get(&url)
+        .query(&exp_context_dimension_value)
+        .headers(headers_map.into())
+        .header(
+            header::AUTHORIZATION,
+            format!("Internal {}", state.superposition_token),
+        )
+        .send()
+        .await;
+
+    process_cac_http_response(response).await
+}
+
+pub async fn get_context_override(
+    user: &User,
+    state: &Data<AppState>,
+    workspace_request: &WorkspaceContext,
+    context_id: String,
+) -> superposition::Result<ContextResp> {
+    let http_client = reqwest::Client::new();
+    let url = state.cac_host.clone() + "/context/" + &context_id;
+    let user_str = serde_json::to_string(user).map_err(|err| {
+        log::error!("Something went wrong, failed to stringify user data {err}");
+        unexpected_error!(
+            "Something went wrong, failed to stringify user data {}",
+            err
+        )
+    })?;
+
+    let extra_headers = vec![("x-user", user_str)];
+
+    let headers_map = construct_header_map(
+        &workspace_request.workspace_id,
+        &workspace_request.organisation_id,
+        extra_headers,
+    )?;
+    let response = http_client
+        .get(&url)
+        .headers(headers_map.into())
+        .header(
+            header::AUTHORIZATION,
+            format!("Internal {}", state.superposition_token),
+        )
+        .send()
+        .await;
+    let resp_contexts = process_cac_http_response(response).await.map_err(|err| {
+        log::error!("Failed to fetch context during cac http call");
+        match err {
+            superposition::AppError::ResponseError(val) => {
+                let mut msg: String =
+                    "failed while checking cac context for given experiment with error:  "
+                        .to_string() + &val.message.clone();
+
+                if val.status_code == StatusCode::NOT_FOUND {
+                    msg = "Context not found in CAC for given experiment, you should discard this experiment".to_string() + &val.message.clone();
+                }
+                response_error!(val.status_code, msg)
```

Comment:
ideally, we should not forward the status code here, an error here, should be an `Internal Server Error` with respect to the API using this, right ?
the only difference being that the message being forwarded is different, just in the case of 404 from this particular API
```suggestion
                match val.status_code {
                   StatusCode::NOT_FOUND => response_error!(503, "Context not found in CAC for given experiment, you should discard this experiment: {}", val.message),
                   _ => err
                }
```